restcomm/restcomm.interpreter/src/test/java/org/restcomm/connect/interpreter/rcml/GatherSpeechTest.java

```diff
@@ -0,0 +1,392 @@
+            GatherAttributes.ATTRIBUTE_PARTIAL_RESULT_CALLBACK + "=\"" + partialCallbackUri + "\" " +
+            GatherAttributes.ATTRIBUTE_NUM_DIGITS + "=\"1\" " +
+            GatherAttributes.ATTRIBUTE_TIME_OUT + "=\"60\">" +
+            "</Gather></Response>";
+    private String gatherEmpty = "<Response><Gather " +
+            GatherAttributes.ATTRIBUTE_PARTIAL_RESULT_CALLBACK + "=\"" + partialCallbackUri + "\">" +
+            "</Gather></Response>";
+
+    public GatherSpeechTest() {
+        super();
+    }
+
+    @BeforeClass
+    public static void before() throws Exception {
+        system = ActorSystem.create();
+    }
+
+    @AfterClass
+    public static void after() throws Exception {
+        system.shutdown();
+    }
+
+    @Before
+    public void init() {
+        String restcommXmlPath = this.getClass().getResource("/restcomm.xml").getFile();
+        try {
+            configuration = getConfiguration(restcommXmlPath);
+            RestcommConfiguration.createOnce(configuration);
+        } catch (ConfigurationException e) {
+            throw new RuntimeException();
+        }
+    }
+
+    private Configuration getConfiguration(String path) throws ConfigurationException {
+        XMLConfiguration xmlConfiguration = new XMLConfiguration();
+        xmlConfiguration.setDelimiterParsingDisabled(true);
+        xmlConfiguration.setAttributeSplittingDisabled(true);
+        xmlConfiguration.load(path);
+
+        return xmlConfiguration;
+    }
+
+    private HttpResponseDescriptor getOkRcml(URI uri, String rcml) {
+        HttpResponseDescriptor.Builder builder = HttpResponseDescriptor.builder();
+        builder.setURI(uri);
+        builder.setStatusCode(200);
+        builder.setStatusDescription("OK");
+        builder.setContent(rcml);
+        builder.setContentLength(rcml.length());
+        builder.setContentType("text/xml");
+        return builder.build();
+    }
+
+    private ActorRef createVoiceInterpreter(final ActorRef observer) {
+        //dao
+        final CallDetailRecordsDao recordsDao = Mockito.mock(CallDetailRecordsDao.class);
+        Mockito.when(recordsDao.getCallDetailRecord(Mockito.any(Sid.class))).thenReturn(null);
+
+        final DaoManager storage = Mockito.mock(DaoManager.class);
+        Mockito.when(storage.getCallDetailRecordsDao()).thenReturn(recordsDao);
+
+        //actors
+        final ActorRef downloader = new MockedActor("downloader")
+                .add(DiskCacheRequest.class, new DiskCacheRequestProperty(playUri), new DiskCacheResponse(playUri))
+                .asRef(system);
+
+        final ActorRef callManager = new MockedActor("callManager").asRef(system);
+
+        final Props props = new Props(new UntypedActorFactory() {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public Actor create() throws Exception {
+                return new VoiceInterpreter(configuration,
```

Comment:
@YevgenL The VoiceInterpreter constructor has changed and this needs to be fixed here